### PR TITLE
Add various DOM and Web tests.

### PR DESCRIPTION
**Pull request by jellefoks**
b/312288290

### CODE REVIEW

File: cobalt/dom/html_element_style_test.cc [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+namespace {
+std::string GetStyleName(::testing::TestParamInfo<const char*> info) {
+  return std::string(info.param);
+}
+
+std::string CamelCaseToSnakeCase(const char* camel_case) {
+  std::string snake;
+  for (const char* input = camel_case; *input; ++input) {
+    if (*input >= 'A' && *input <= 'Z') {
+      snake.push_back('-');
```
Comment (oxve) [inline cobalt/dom/html_element_style_test.cc:46]: `'-'` -> `'_'`

Comment (jellefoks) [reply cobalt/dom/html_element_style_test.cc:46]: It's the css property name.

File: cobalt/persistent_storage/persistent_settings.cc [hunk 2]
```diff
@@ -48,7 +48,8 @@ PersistentSettings::PersistentSettings(const std::string& file_name)
                   kSbFileMaxPath);
   persistent_settings_file_ =
       std::string(storage_dir.data()) + kSbFileSepString + file_name;
-  LOG(INFO) << "Persistent settings file path: " << persistent_settings_file_;
+  // LOG(INFO) << "Persistent settings file path: " <<
```
Comment (oxve) [inline cobalt/persistent_storage/persistent_settings.cc:51]: Did you mean to comment this out?

File: starboard/common/file.cc [hunk 3]
```diff
@@ -50,7 +50,7 @@ void RecordFileWriteStat(int write_file_result) {
 
 bool SbFileDeleteRecursive(const char* path, bool preserve_root) {
   if (!SbFileExists(path)) {
-    SB_LOG(ERROR) << "Path does not exist: '" << path << "'";
+    // SB_LOG(ERROR) << "Path does not exist: '" << path << "'";
```
Comment (oxve) [inline starboard/common/file.cc:53]: Did you mean to leave this commented out? Remove?

**Review by oxve — APPROVED** [review]
Three minor comments

File: cobalt/dom/html_element_style_test.cc [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+namespace dom {
+
+using script::testing::FakeScriptValue;
+
+namespace {
+std::string GetStyleName(::testing::TestParamInfo<const char*> info) {
+  return std::string(info.param);
+}
+
+std::string CamelCaseToSnakeCase(const char* camel_case) {
```
[the remark this answers is not in the document]
Comment (jellefoks) [reply cobalt/dom/html_element_style_test.cc:42]: Done

**Pull request merged**